### PR TITLE
[Quest/Malawi Core] Android App : Sync Inactive recently changed Resources which are being filtered

### LINKED ISSUE #73: [Quest/Malawi Core] Android App : Sync Inactive recently changed Resources which are being filtered
**Description of feature**

Sync Inactive recently changed Resources which are being filtered

### PR DESCRIPTION
**Pull request by calmwalija**
Fixes #73

**Engineer Checklist**
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by sevenreup** [comment]
@BuchiNy can redo the tests you did on syncing inactive resources on this branch, especially with patients

File: android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncListenerManager.kt [hunk 1]
```diff
@@ -53,6 +53,12 @@ constructor(
 
   private val syncConfig by lazy { configurationRegistry.getSyncConfigs() }
 
+  private fun isInitialSync() =
+    sharedPreferencesHelper
+      .read(SharedPreferenceKey.LAST_SYNC_TIMESTAMP.name, null)
+      .isNullOrBlank()
+      .not()
```
Comment (sevenreup) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncListenerManager.kt]: Is initial sync is when the timestamp is null

https://github.com/d-tree-org/fhircore/blob/b5b91af98ed6fbc818d76b6ec71632ed2e35b13b/android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt#L112-L113

**Review by sevenreup — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncListenerManager.kt (hunk 1 above)
Comment (calmwalija) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncListenerManager.kt]: ### I don't understand what this means
> Is initial sync is when the timestamp is null

Comment (sevenreup) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncListenerManager.kt]: The timestamp is set after the initial sync, so if you want to check if a sync is an initial sync you have to check if the timestamp is null. On that line, you are negating which means this is more of a check if the sync is subsequent than the initial one. Check the implementation in `SyncBroadcaster.kt`

**Review by BuchiNy — APPROVED** [review]

**Pull request merged**